plugins/mcreator-core/blockly/js/mcreator_extensions.js

```diff
@@ -87,6 +87,42 @@ Blockly.Extensions.registerMutator('variable_entity_input',
         }
     });
 
+// Helper function to use in Blockly extensions that register one data list selector field to update contents of another
+// The block may define input called "<targetName>Field" to customize field's position
+// Note that the source field must be inserted before the target field for their values to be loaded properly
+function appendAutoReloadingDataListField(sourceName, targetName, targetList) {
+    return function () {
+        const thisBlock = this;
+        (this.getInput(targetName + 'Field') || this.appendDummyInput()).appendField(
+            new FieldDataListSelector(targetList, undefined, {
+                'customEntryProviders': function () {
+                    return thisBlock.getFieldValue(sourceName);
+                }
+            }), targetName);
+        this.setOnChange(function (changeEvent) {
+            if (changeEvent.type === Blockly.Events.BLOCK_CHANGE &&
+                changeEvent.group && changeEvent.blockId === this.id &&

```

Comment:
A comment explaining need for "changeEvent.group" would also fit, maybe above if statement.

It is all but obvious for external observer why this is there

Reply:
✔️ Fixed